From c07df4123f8d73ea5f106a5fd2c7e7437f3d75f7 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 17 Apr 2025 21:41:31 -0400
Subject: [PATCH] Support supervanilla flag

---
 bliss_prebuilts.mk | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/bliss_prebuilts.mk b/bliss_prebuilts.mk
index 8ca70b9..285061e 100644
--- a/bliss_prebuilts.mk
+++ b/bliss_prebuilts.mk
@@ -1,6 +1,12 @@
 include $(call first-makefiles-under,$(LOCAL_PATH))
 
+ifneq ($(BLISS_SUPER_VANILLA),true)
+
 PRODUCT_PACKAGES += \
      yetCalc \
-     AboutBliss \
-     Jellyfish
+     AboutBliss 
+
+endif
+
+PRODUCT_PACKAGES += \
+     Jellyfish
\ No newline at end of file
-- 
2.43.0

