From a0f63cc1f85a0d02cc30ca115421679bec957048 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 11 Jan 2025 17:08:59 -0500
Subject: [PATCH 01/42] include bass_init.rc into init.x86.rc

---
 init.x86.rc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init.x86.rc b/init.x86.rc
index 4c495c7..0220b3b 100644
--- a/init.x86.rc
+++ b/init.x86.rc
@@ -1,4 +1,5 @@
 import /vendor/etc/init/hw/init.low_performance.rc
+import /vendor/etc/init/bass_init.rc
 
 on early-init
     write /proc/sys/kernel/ctrl-alt-del 1
-- 
2.43.0

