From 8648e72386ee5a1cd67cceb785699ead7db94f72 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 30 Jan 2025 16:20:10 -0500
Subject: [PATCH 08/18] bass: set desktop mode on boot

---
 device.mk | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/device.mk b/device.mk
index ccc4e14..e6b1b37 100644
--- a/device.mk
+++ b/device.mk
@@ -313,8 +313,7 @@ $(call inherit-product-if-exists, vendor/bass/branding.mk)
 
 # Enable MultiWindow
 PRODUCT_PROPERTY_OVERRIDES += \
-    persist.sys.debug.multi_window=true \
-    persist.sys.debug.desktop_mode=true
+    persist.sys.debug.multi_window=true
 
 # DRM service opt-in
 PRODUCT_VENDOR_PROPERTIES += drm.service.enabled=true
-- 
2.43.0

