From 098c1a8b6be60e04fb4ce8f6011e41fed1fbfcc1 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 30 Jan 2025 16:20:10 -0500
Subject: [PATCH 18/42] bass: set desktop mode on boot

---
 device.mk | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/device.mk b/device.mk
index 539cef4..ab160f8 100644
--- a/device.mk
+++ b/device.mk
@@ -315,11 +315,19 @@ $(call inherit-product-if-exists, vendor/boringdroid/boringdroid.mk)
 # Add Bass Toolkit
 $(call inherit-product-if-exists, vendor/bass/branding.mk)
 
+ifneq ($(or $(USE_BLISS_KIOSK_LAUNCHER), $(USE_BLISS_RESTRICTED_LAUNCHER)),true)
 # Enable MultiWindow
 PRODUCT_PROPERTY_OVERRIDES += \
     persist.sys.debug.multi_window=true
+
+ifeq ($(USE_DESKTOP_MODE_ON_SECONDARY_DISPLAY),true)
+PRODUCT_PROPERTY_OVERRIDES += \
     persist.sys.debug.desktop_mode=true
 
+endif # USE_DESKTOP_MODE_ON_SECONDARY_DISPLAY
+
+endif
+
 # DRM service opt-in
 PRODUCT_VENDOR_PROPERTIES += drm.service.enabled=true
 
-- 
2.43.0

