From e5b34eaefeb1766ccd77b365fdc57cce0e2f38f8 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 31 May 2024 22:46:22 -0400
Subject: [PATCH 01/23] fix: enable desktop_mode by default

---
 device.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/device.mk b/device.mk
index 4da3237..15b1cb1 100644
--- a/device.mk
+++ b/device.mk
@@ -312,7 +312,7 @@ $(call inherit-product-if-exists, vendor/boringdroid/boringdroid.mk)
 
 # Enable MultiWindow
 PRODUCT_PROPERTY_OVERRIDES += \
-    persist.sys.debug.multi_window=true
+    persist.sys.debug.multi_window=true \
     persist.sys.debug.desktop_mode=true
 
 # DRM service opt-in
-- 
2.34.1

