From 0cc84a968bf699ba7a52631d42499c8a9efb76ac Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Wed, 29 May 2024 12:39:48 -0400
Subject: [PATCH 05/10] inherit bass vendor functions

---
 build/envsetup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/envsetup.sh b/build/envsetup.sh
index 682353fd..d452a071 100644
--- a/build/envsetup.sh
+++ b/build/envsetup.sh
@@ -1078,6 +1078,8 @@ function blissify()
     return $?
 }
 
+source vendor/bass/bass_setup.sh
+
 # Override host metadata to make builds more reproducible and avoid leaking info
 export BUILD_USERNAME=nobody
 export BUILD_HOSTNAME=android-build
-- 
2.43.0

