From 9d6c5cc95f2a2c156c052ffdd3810807c319a1ee Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Tue, 15 Oct 2024 12:12:56 -0400
Subject: [PATCH 06/10] Call bass_build_config on envsetup

---
 build/envsetup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/envsetup.sh b/build/envsetup.sh
index d452a071..fa937f20 100644
--- a/build/envsetup.sh
+++ b/build/envsetup.sh
@@ -1080,6 +1080,8 @@ function blissify()
 
 source vendor/bass/bass_setup.sh
 
+bass_build_config
+
 # Override host metadata to make builds more reproducible and avoid leaking info
 export BUILD_USERNAME=nobody
 export BUILD_HOSTNAME=android-build
-- 
2.43.0

