From 02fbd9ac8f99927f2d54fce24aa90266ade75703 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 18 Jul 2024 12:58:52 -0400
Subject: [PATCH 1/2] linker: updates for x86 & x86_64 linking

---
 hybris/common/q/linker.cpp | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/hybris/common/q/linker.cpp b/hybris/common/q/linker.cpp
index dcc8428..8006a43 100644
--- a/hybris/common/q/linker.cpp
+++ b/hybris/common/q/linker.cpp
@@ -139,6 +139,35 @@ static const char* const kRuntimeApexLibDir   = "/apex/com.android.runtime/lib";
 static const char* const kI18nApexLibDir      = "/apex/com.android.i18n/lib";
 #endif
 
+#if defined(__x86_64__) || defined(_M_X64)
+static const char* const kSystemLibDir        = "/system/lib64";
+static const char* const kSystemExtLibDir     = "/system/system_ext/lib64";
+static const char* const kOdmLibDir           = "/odm/lib64";
+static const char* const kVendorLibDir        = "/system/vendor/lib64";
+static const char* const kVendorEglLibDir     = "/system/vendor/lib64/egl";
+static const char* const kVendorHwLibDir      = "/system/vendor/lib64/hw";
+static const char* const kProductLibDir       = "/system/product/lib64";
+static const char* const kAsanSystemLibDir    = "/data/asan/system/lib64";
+static const char* const kAsanOdmLibDir       = "/data/asan/odm/lib64";
+static const char* const kAsanVendorLibDir    = "/data/asan/vendor/lib64";
+static const char* const kRuntimeApexLibDir   = "/apex/com.android.runtime/lib64";
+static const char* const kI18nApexLibDir      = "/apex/com.android.i18n/lib64";
+#endif
+#if defined(i386) || defined(__i386__) || defined(__i386) || defined(_M_IX86)
+static const char* const kSystemLibDir        = "/system/lib";
+static const char* const kSystemExtLibDir     = "/system/system_ext/lib";
+static const char* const kOdmLibDir           = "/odm/lib";
+static const char* const kVendorLibDir        = "/system/vendor/lib";
+static const char* const kVendorEglLibDir     = "/system/vendor/lib/egl";
+static const char* const kVendorHwLibDir      = "/system/vendor/lib/hw";
+static const char* const kProductLibDir       = "/system/product/lib";
+static const char* const kAsanSystemLibDir    = "/data/asan/system/lib";
+static const char* const kAsanOdmLibDir       = "/data/asan/odm/lib";
+static const char* const kAsanVendorLibDir    = "/data/asan/vendor/lib";
+static const char* const kRuntimeApexLibDir   = "/apex/com.android.runtime/lib";
+static const char* const kI18nApexLibDir      = "/apex/com.android.i18n/lib";
+#endif
+
 static const char* const kAsanLibDirPrefix = "/data/asan";
 
 static const char* const kDefaultLdPaths[] = {
-- 
2.34.1

