From 77e29114ad0c7f43220867ee7dcef365e666a8ac Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Wed, 14 May 2025 19:21:51 -0400
Subject: [PATCH] use a switch to skip aaropa download

---
 vendorsetup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vendorsetup.sh b/vendorsetup.sh
index 804f919..5399e32 100755
--- a/vendorsetup.sh
+++ b/vendorsetup.sh
@@ -13,4 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+if [ ! "${SKIP_AAROPA_DOWNLOAD}" = "true" ]; then
 bash bootable/aaropa/download.sh
+fi
-- 
2.43.0

