From 2f0d0d37a2c7532839693183a7fe40cd82e10c42 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 30 May 2024 10:55:43 -0400
Subject: [PATCH 2/2] soong: whitelist bindgen

---
 ui/build/paths/config.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ui/build/paths/config.go b/ui/build/paths/config.go
index b365206d7..a5e7163da 100644
--- a/ui/build/paths/config.go
+++ b/ui/build/paths/config.go
@@ -88,6 +88,7 @@ func GetConfig(name string) PathConfig {
 var Configuration = map[string]PathConfig{
 	"bash":    Allowed,
         "cpio":    Allowed,
+	"bindgen":   Allowed,
 	"bison":   Allowed,
 	"brotli":  Allowed,
 	"ccache":  Allowed,
-- 
2.34.1

