From f0e9339ca81522a96ce8c98633a183050b32e013 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 7 Jun 2024 10:20:18 -0400
Subject: [PATCH 15/18] rework lockdown app installs

---
 .../android/packageinstaller/PackageInstallerActivity.java  | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/packages/PackageInstaller/src/com/android/packageinstaller/PackageInstallerActivity.java b/packages/PackageInstaller/src/com/android/packageinstaller/PackageInstallerActivity.java
index bb1ad1aa144c..43371197828d 100644
--- a/packages/PackageInstaller/src/com/android/packageinstaller/PackageInstallerActivity.java
+++ b/packages/PackageInstaller/src/com/android/packageinstaller/PackageInstallerActivity.java
@@ -518,11 +518,7 @@ public class PackageInstallerActivity extends AlertActivity {
      * Check if it is allowed to install the package and initiate install if allowed.
      */
     private void checkIfAllowedAndInitiateInstall() {
-        if (SystemProperties.get("ro.boot.bliss.bootmode").equals("lockdown")) {
-            showDialogInner(DLG_UNKNOWN_SOURCES_RESTRICTED_FOR_USER);
-            return;
-        }
-        if (mAllowUnknownSources || !isInstallRequestFromUnknownSource(getIntent())) {
+        if (mAllowUnknownSources || !isInstallRequestFromUnknownSource(getIntent()) || !SystemProperties.get("ro.boot.bliss.bootmode").equals("lockdown")) {
             if (mLocalLOGV) Log.i(TAG, "install allowed");
             initiateInstall();
         } else {
-- 
2.34.1

