From 3e9a5569cf54c142578d95afffa6d670eb50b2d3 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Tue, 11 Jun 2024 16:30:10 -0400
Subject: [PATCH 07/20] use the enabled property instead

---
 services/inputflinger/reader/InputDevice.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/services/inputflinger/reader/InputDevice.cpp b/services/inputflinger/reader/InputDevice.cpp
index 4c07738821..7a0766a83e 100644
--- a/services/inputflinger/reader/InputDevice.cpp
+++ b/services/inputflinger/reader/InputDevice.cpp
@@ -267,7 +267,7 @@ void InputDevice::configure(nsecs_t when, const InputReaderConfiguration* config
     char mIgnoreExternalInput[PROPERTY_VALUE_MAX] = {0};
     property_get("ro.boot.force.ignore_external_input", mIgnoreExternalInput, "0");
 
-    if (!isIgnored() || !((strcmp(mIgnoreExternalInput, "1") == 0) && (mIsExternal == true))) {
+    if (!isIgnored()) {
         if (!changes) { // first time only
             mConfiguration.clear();
             for_each_subdevice([this](InputDeviceContext& context) {
@@ -357,6 +357,10 @@ void InputDevice::configure(nsecs_t when, const InputReaderConfiguration* config
                           inputDeviceName.c_str(), mAssociatedDisplayUniqueId->c_str());
                     enabled = false;
                 }
+            } else if (strcmp(mIgnoreExternalInput, "1") == 0) {
+                if (mIsExternal) {
+                    enabled = false;
+                }
             }
 
             if (changes) {
-- 
2.43.0

