From 3dca22525024686a993784e45c34b00c9bbb9737 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 17 Apr 2025 21:41:31 -0400
Subject: [PATCH] Support supervanilla flag

---
 bliss_prebuilts.mk | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/bliss_prebuilts.mk b/bliss_prebuilts.mk
index 761d0af..9283e62 100644
--- a/bliss_prebuilts.mk
+++ b/bliss_prebuilts.mk
@@ -1,8 +1,14 @@
 include $(call first-makefiles-under,$(LOCAL_PATH))
 
+ifneq ($(BLISS_SUPER_VANILLA),true)
+
 PRODUCT_PACKAGES += \
      yetCalc \
      AboutBliss \
      Gramophone \
-     Jellyfish \
      Lawnicons
+
+endif
+
+PRODUCT_PACKAGES += \
+     Jellyfish
\ No newline at end of file
-- 
2.43.0

