From 390416ae8a5588a16997dd5b9fa8a1e422fd38d7 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 7 Mar 2024 09:37:19 -0500
Subject: [PATCH 1/5] Update .gitignore for bass additions

---
 .gitignore | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.gitignore b/.gitignore
index 931ce07a..1188a8e0 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,5 @@
 proprietary/*
 prebuilt/archive
+blissethernet/*
+config/watermark/*
+config/signing/*
\ No newline at end of file
-- 
2.34.1

