From e31c5279dbe2d92db68a31211e9cb46095796e5e Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Wed, 29 May 2024 12:39:48 -0400
Subject: [PATCH 5/5] inherit bass vendor functions

---
 build/envsetup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/envsetup.sh b/build/envsetup.sh
index 29dd0a47..a371008f 100644
--- a/build/envsetup.sh
+++ b/build/envsetup.sh
@@ -1088,6 +1088,8 @@ function blissify()
     return $?
 }
 
+source vendor/bass/bass_setup.sh
+
 # Override host metadata to make builds more reproducible and avoid leaking info
 export BUILD_USERNAME=nobody
 export BUILD_HOSTNAME=android-build
-- 
2.34.1

