From a14261afbde1203e7158193e5522b7d671cc1342 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 31 May 2024 22:49:00 -0400
Subject: [PATCH 03/25] Inherit bass toolkit

---
 device.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/device.mk b/device.mk
index ad17e85..b7cd12f 100644
--- a/device.mk
+++ b/device.mk
@@ -306,6 +306,9 @@ $(call inherit-product-if-exists, vendor/google/proprietary/SettingsIntelligence
 # Boringdroid
 $(call inherit-product-if-exists, vendor/boringdroid/boringdroid.mk)
 
+# Add Bass Toolkit
+$(call inherit-product-if-exists, vendor/bass/branding.mk)
+
 # Enable MultiWindow
 PRODUCT_PROPERTY_OVERRIDES += \
     persist.sys.debug.multi_window=true \
-- 
2.34.1

