From 2007aa95694e308f769353182c4f20c1dd75abd8 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 12 Oct 2024 17:45:27 -0400
Subject: [PATCH 04/23] init: Add set_max_logd options

---
 init.sh | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/init.sh b/init.sh
index 48cbc08..eb66122 100644
--- a/init.sh
+++ b/init.sh
@@ -976,6 +976,40 @@ function set_usb_mode()
 	done
 }
 
+function set_max_logd()
+{
+	for c in `cat /proc/cmdline`; do
+		case $c in
+			*=*)
+				eval $c
+				if [ -z "$1" ]; then
+					case $c in
+						# Set TimeZone
+						SET_MAX_LOGD=*)
+							if [ "$SET_MAX_LOGD" == 1 ]; then
+								size_value="8388608"
+								radio_size_value="4M"
+								system_size_value="4M"
+								crash_size_value="1M"
+							else
+								size_value=""
+								radio_size_value=""
+								system_size_value=""
+								crash_size_value=""
+							fi
+							setprop persist.logd.size "$size_value"
+							setprop persist.logd.size.radio "$radio_size_value"
+							setprop persist.logd.size.system "$system_size_value"
+							setprop persist.logd.size.crash "$crash_size_value"
+							;;
+					esac
+				fi
+				;;
+		esac
+	done
+	
+}
+
 function do_init()
 {
 	init_misc
@@ -984,6 +1018,7 @@ function do_init()
 	set_custom_timezone
 	init_hal_audio
 	set_custom_ota
+	set_max_logd
 	init_hal_brcm_wifi
 	init_hal_bluetooth
 	init_hal_camera
-- 
2.34.1

