From 9d49cea394453a785041cb111446097b9cb1b140 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 31 May 2024 22:57:46 -0400
Subject: [PATCH 05/23] init: Add set_iio_options options...

---
 init.sh | 41 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/init.sh b/init.sh
index eb66122..0998867 100644
--- a/init.sh
+++ b/init.sh
@@ -1010,6 +1010,46 @@ function set_max_logd()
 	
 }
 
+# Allow iio options to be passed through cmdline
+function set_iio_options()
+{
+	# Set up usb/adb props when values are detected in /proc/cmdline
+	
+	for c in `cat /proc/cmdline`; do
+		case $c in
+			*=*)
+				eval $c
+				if [ -z "$1" ]; then
+					case $c in
+						SET_IIO_ORDER=*)
+							set_property ro.iio.accel.order "$SET_IIO_ORDER"
+							;;
+						SET_IIO_ACCEL_QUIRKS=*)
+							set_property ro.iio.accel.quirks "$SET_IIO_ACCEL_QUIRKS"
+							;;
+						SET_IIO_ACCEL_X_OPT_SCALE=*)
+							set_property ro.iio.accel.x.opt_scale "$SET_IIO_ACCEL_X_OPT_SCALE"
+							;;
+						SET_IIO_ACCEL_Y_OPT_SCALE=*)
+							set_property ro.iio.accel.y.opt_scale "$SET_IIO_ACCEL_Y_OPT_SCALE"
+							;;
+						SET_IIO_ANGLVEL_QUIRKS=*)
+							set_property ro.iio.anglvel.quirks "$SET_IIO_ANGLVEL_QUIRKS"
+							;;
+						SET_IGNORE_ATKBD=*)
+							# (0,1) 0=off, 1=on
+							set_property ro.ignore_atkbd "$SET_IGNORE_ATKBD"
+							;;
+						SET_IIO_MAGN_QUIRKS=*)
+							set_property ro.iio.magn.quirks "$SET_IIO_MAGN_QUIRKS"
+							;;
+					esac
+				fi
+				;;
+		esac
+	done
+}
+
 function do_init()
 {
 	init_misc
@@ -1032,6 +1072,7 @@ function do_init()
 	init_hal_thermal
 	init_hal_sensors
 	init_hal_surface
+	set_iio_options
 	init_tscal
 	init_ril
 	init_loop_links
-- 
2.34.1

