From ce1732f604f608a602551ec1a1f0bcaa83f97719 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 12 Oct 2024 17:47:06 -0400
Subject: [PATCH 09/23] init: add init_misc_cpu options

---
 init.sh | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/init.sh b/init.sh
index 1186820..189c5ac 100644
--- a/init.sh
+++ b/init.sh
@@ -30,6 +30,8 @@ function init_misc()
 	# in case no cpu governor driver autoloads
 	[ -d /sys/devices/system/cpu/cpu0/cpufreq ] || modprobe acpi-cpufreq
 
+	init_misc_cpu
+
 	# enable sdcardfs/esdfs if /data is not mounted on tmpfs or 9p
 	mount | grep /data\ | grep -qE 'tmpfs|9p'
 	[ $? -eq 0 ] && SDCARDFS_DISABLE=${SDCARDFS_DISABLE:-1}
@@ -1436,6 +1438,40 @@ function set_package_opts()
 	
 }
 
+function init_misc_cpu()
+{
+	# Allow for adjusting intel_pstate max/min freq on boot
+	if [ -n "$INTEL_PSTATE_CPU_MIN_PERF_PCT"  ]; then
+		echo $INTEL_PSTATE_CPU_MIN_PERF_PCT > /sys/devices/system/cpu/intel_pstate/min_perf_pct
+	fi
+	
+	if [ -n "$INTEL_PSTATE_CPU_MAX_PERF_PCT"  ]; then
+		echo $INTEL_PSTATE_CPU_MAX_PERF_PCT > /sys/devices/system/cpu/intel_pstate/max_perf_pct
+	fi
+
+	# Allow for adjusting cpu energy performance
+	# Normal options: default, performance, balance_performance, balance_power, power
+	if [ -n "$CPU_ENERGY_PERFORMANCE_PREF"  ]; then
+
+		cpuprefavailable=$(cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences)
+		
+		# if the word "CPU_ENERGY_PERFORMANCE_PREF" is found in the $cpuprefavailable string, then set the options for all cpus
+		if [[ "$cpuprefavailable" == *"$CPU_ENERGY_PERFORMANCE_PREF"* ]]; then
+			for cpu in $(ls -d /sys/devices/system/cpu/cpu?); do
+				echo $CPU_ENERGY_PERFORMANCE_PREF > $cpu/cpufreq/energy_performance_preference || return 1
+			done
+		fi
+
+	fi
+
+	# Set cpu scaling at boot (ie set pstate status to active/passive at boot)
+	if [ -n "$INTEL_PSTATE_STATUS"  ]; then
+		echo $INTEL_PSTATE_STATUS > /sys/devices/system/cpu/intel_pstate/status
+	else
+		echo passive > /sys/devices/system/cpu/intel_pstate/status
+	fi
+}
+
 function do_init()
 {
 	init_misc
-- 
2.34.1

