From 2edf12090c0c26702620b08392ddc33b4df846c6 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 31 May 2024 23:11:36 -0400
Subject: [PATCH 10/23] init: add TAIFAElimuTab sleep override

---
 init.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init.sh b/init.sh
index 189c5ac..ce93b3a 100644
--- a/init.sh
+++ b/init.sh
@@ -626,6 +626,9 @@ function init_hal_power()
 		e-tab*Pro)
 			SLEEP_STATE=force
 			;;
+		*TAIFAElimuTab*)
+			setprop sleep.earlysuspend 1
+			;;
 		*)
 			;;
 	esac
-- 
2.34.1

