From 48da6578b4fe8cfe39db8eafa78884dedb357118 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 31 May 2024 23:13:05 -0400
Subject: [PATCH 11/23] init: better sleep state handling

---
 init.sh | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/init.sh b/init.sh
index ce93b3a..d7f3335 100644
--- a/init.sh
+++ b/init.sh
@@ -633,7 +633,18 @@ function init_hal_power()
 			;;
 	esac
 
-	set_property sleep.state ${SLEEP_STATE}
+	# Detect is deep sleep is supported and auto set sleep.state parameter
+	mem_sleep_default=$(cat /sys/power/mem_sleep)
+	if [[ "$mem_sleep_default" == *deep* ]]; then
+		setprop sleep.state ${SLEEP_STATE:-mem}
+	else
+		setprop sleep.state ${SLEEP_STATE:-freeze}
+	fi
+
+	current_sleep_state=$(getprop sleep.state)
+	if [ "${current_sleep_state}" = "mem" ]; then
+		echo "deep" > /sys/power/mem_sleep
+	fi
 }
 
 function init_hal_thermal()
-- 
2.34.1

