From a277b290d7881438d548e2a1c69828df7120224c Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 27 Jun 2024 11:46:41 -0400
Subject: [PATCH 17/23] Use fuse passthrough

---
 device.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/device.mk b/device.mk
index 642dd73..eee14de 100644
--- a/device.mk
+++ b/device.mk
@@ -70,6 +70,10 @@ PRODUCT_PRODUCT_PROPERTIES += \
     ro.lmk.use_new_strategy=false
 endif
 
+# Use FUSE passthrough
+PRODUCT_PRODUCT_PROPERTIES += \
+    persist.sys.fuse.passthrough.enable=true
+
 PRODUCT_COPY_FILES := \
     $(if $(wildcard $(PRODUCT_DIR)init.rc),$(PRODUCT_DIR)init.rc:root/init.rc) \
     $(if $(wildcard $(PRODUCT_DIR)init.sh),$(PRODUCT_DIR),$(LOCAL_PATH)/)init.sh:system/etc/init.sh \
@@ -179,7 +183,7 @@ $(foreach f,$(wildcard $(LOCAL_PATH)/permissions/*.xml),\
     $(eval PRODUCT_COPY_FILES += $(f):$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/permissions/$(notdir $f)))
 
 # Get emulated storage settings
-#$(call inherit-product, $(SRC_TARGET_DIR)/product/emulated_storage.mk)
+$(call inherit-product, $(SRC_TARGET_DIR)/product/emulated_storage.mk)
 
 # Get Android 8.0 HIDL HALs
 $(call inherit-product,$(LOCAL_PATH)/treble.mk)
-- 
2.34.1

