From 03ca264a7829c36c3b2f765a4a782c04c98e9be1 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 10 Apr 2025 13:08:11 -0400
Subject: [PATCH 5/5] init: give RCVRY some debugging features

---
 initrd/init | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/initrd/init b/initrd/init
index 7eda3c2..7be5d51 100755
--- a/initrd/init
+++ b/initrd/init
@@ -199,7 +199,7 @@ EOF
 }
 
 # configure debugging output
-if [ "$DEBUG" ] || [ "$INSTALL" ]; then
+if [ "$DEBUG" ] || [ "$INSTALL" ] || [ "$RCVRY" ]; then
 	LOG=/tmp/log
 	set -x
 else
@@ -226,7 +226,7 @@ if test x"$HAS_CTTY" != x"Yes"; then
 		mknod /dev/tty2 c 4 2 && openvt
 		mknod /dev/tty3 c 4 3 && openvt
 	fi
-	if [ "$INSTALL" ] || [ "$DEBUG" ]; then
+	if [ "$INSTALL" ] || [ "$DEBUG" ] || [ "$RCVRY" ]; then
 		echo 0 0 0 0 >/proc/sys/kernel/printk
 	fi
 	# initialise /dev (first time)
-- 
2.43.0

